### PR TITLE
Add cranelift meeting notes

### PR DESCRIPTION
**Pull request by alexcrichton**

### CODE REVIEW

File: meetings/cranelift/2021/cranelift-10-04.md [hunk 1]
```diff
@@ -14,8 +14,155 @@
...
+
+... alexcrichton had video issues and disconnected ...
+
+* cfallin - if we have few weird special cases easier to consume than a more
+  general form.
+* bbouvier - main concern is about the developer experience. The systems seem
+  reasonable. Having a very simple tutorial is a great thing though. Seems great
+  at solving 95% of the problems and for the remaining 5% you need to be an
+  expert (meaning writing your own extractor), but maybe this is the right
+  tradeoff? It's a regression to deal with two kinds and not see how they
```
Comment (bnjbvr) [inline meetings/cranelift/2021/cranelift-10-04.md:103]:
```suggestion
  tradeoff? It's a regression to deal with two languages and not see how they
```

File: meetings/cranelift/2021/cranelift-10-04.md [hunk 2]
```diff
@@ -14,8 +14,155 @@
...
+  folks would find that useful. Shows some code generated. After that some open
+  discussion?
+
+... slides! ...
+
+* cfallin - concerns after presentation?
+* uweigand - not really a concern but need details like special cases and
+  particular constants in particular ranges. Can always do whatever since it's
+  custom Rust. Always cases to do something special.
+* cfalling - special cases are extractors and constructors -- becomes a trait
```
Comment (cfallin) [inline meetings/cranelift/2021/cranelift-10-04.md:46]: s/cfalling/cfallin/ (I am sometimes a victim of gravity but not in this case)

File: meetings/cranelift/2021/cranelift-10-04.md [hunk 3]
```diff
@@ -14,8 +14,155 @@
...
+  as well yeah.
+* fitzgen - I think this is a non-issue since we already have `iadd_imm`
+  canonicalized by the preopt pass, no need to re-canonicalize in lowerings.
+* cfallin - still good question if we can express a family of patterns. We could
+  have a macro system perhaps in the futures where one rule goes to N rules, but
+  maybe too much complexity too. For now cases should be simple enough.
+* fitzgen - gcc/go have macro in their DSL to be polymorphic over bit-width.
+* cfallin - for that specific case we could
+
+... alexcrichton had video issues and disconnected ...
```
Comment (cfallin) [inline meetings/cranelift/2021/cranelift-10-04.md:95]: I think this was just a note about how we could be polymorphic on types by binding the type with a pattern variable.

**Review by cfallin — APPROVED** [review]
Thanks for taking such detailed notes!

**Pull request merged**